ast/compile_test.go

```diff
@@ -3661,7 +3666,7 @@ func TestCompilerBuildComprehensionIndexKeySet(t *testing.T) {
 			wantDebug: 1,
 		},
 		{
-			note: "skip: due to nested comprehension containing candidate",
+			note: "skip: due to nested comprehension containing candidate + indexed nested comprehension with key from rule body",
```

Comment:
nit: "skip:" was intended for cases where no index gets built because it produce incorrect results, worsen performance, etc. Let's use a difference prefix ("mixed:"?) since an index is built now.

Reply:
nit: I'd also update the comment in the module that says _'y' disqualifies indexing ..._ to be a bit less ambiguous... it disqualifies indexing of the outer comprehension.